types.ts

```diff
@@ -55,6 +55,12 @@ export enum AuthenticationType {
    * @see {@link CustomHeaderTokenAuthentication}
    */
   CustomHeaderToken = 'CustomHeaderToken',
```

Comment:
Not super important, but a question: do we plan to deprecate `CustomHeaderToken` and/or make `MultiHeaderToken` require at least one header?

If `MultiHeaderToken` can do everything `CustomHeaderToken` can, seems like there's a possible argument to consolidate

Reply:
Yeah, I think it might be reasonable to deprecated CustomHeaderToken. It is functionally redundant now. It is easier to use for the novice though, but maybe not so much to justify 3 flavors of header auth.

Reply:
Hypothetically if we were thinking of consolidating, it might make sense to use a name that doesn't have `Multi` in it if it can support a single header just as well

I'm not proposing any name changes here, but figured it was a thought worth mentioning

Reply:
We could even go a step further and have a `Tokens` auth, that allows you to prompt for any number of query and/or header tokens. Not something we necessarily should do now, but there's probably an even wider solution space to consider.

Reply:
Yeah, all fair, had been thinking about this too. I think if `CustomHeaderToken` wasn't already taken I would have used that or something similar, so this was kind of an effort to distinguish these.

What do people think about `GenericHeaderToken` or `GenericHeaderToken`, or is that too similar to `CustomHeaderToken`?

Reply:
@dweitzman-codaio @ekoleda-codaio Thoughts on `GenericHeaderToken` vs `MultiHeaderToken`?

Reply:
Would it be crazy to reuse `AuthenticationType.CustomHeaderToken` and just have a different `BaseAuthentication` subclass so that either of these would work:

```
// Old version, implies type `CustomHeaderTokenAuthentication`
pack.setUserAuthentication({
  type: coda.AuthenticationType.CustomHeaderToken,
  headerName: "Authorization",
  tokenPrefix: "SSWS",
})

// New version, implies the new type `MultiHeaderTokenAuthentication` but has the same `AuthenticationType` value
pack.setUserAuthentication({
  type: coda.AuthenticationType.CustomHeaderToken,
  headers: [{
    name: "Authorization",
    description: "Foo",
    tokenPrefix: "SSWS",
  }]
})
```

It'd make our internal code more more complex since we'd have to check for `auth.headers` vs `auth.headerName` to know what we're dealing with, but it might keep the SDK surface simpler

Reply:
It's not a completely crazy idea. But I think breaking the use of a strict discriminated union would involve some amount of pain for us in code, for makers, and for documentation.

Where we're at right now is definitely goofy, but I'm not sure this case is common enough to bend over backwards for cleanliness.

Reply:
That's fair.

Since `GenericHeaderToken` feels inconsistent with existing naming and `CustomHeaderTokens` would be confusing since `CustomHeaderToken` already exists, I'm thinking that maybe having both `MultiHeaderToken` and `CustomHeaderToken` exist separately could be the least awkward option.